mlflow/tensorflow.py

```diff
@@ -1058,11 +1064,16 @@ def add_summary(self, *args, **kwargs):
         (EventFileWriterV2, "add_event", add_event),
         (tensorflow.estimator.Estimator, "train", train),
         (tensorflow.keras.Model, "fit", fit),
-        (tensorflow.keras.Model, "fit_generator", fit_generator),
         (tensorflow.estimator.Estimator, "export_saved_model", export_saved_model),
         (tensorflow.estimator.Estimator, "export_savedmodel", export_savedmodel),
         (FileWriter, "add_summary", add_summary),
     ]
+    if LooseVersion(tensorflow.__version__) <= LooseVersion("2.0.0"):
+        # `fit_generator()` is deprecated in TF > 2.0.0 and simply wraps `fit()`.
```

Comment:
Ideally, in the future, we will make use of a utility similar to scikit-learn's `_SklearnTrainingSession` to guard against nested runs in other ML autologging integrations. For now, this appears to be a quick & perfectly valid fix.